tests/flows/erc20_to_native_token_bridge.go

```diff
@@ -0,0 +1,364 @@
+package flows
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	erc20tokensource "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/ERC20TokenSource"
+	nativetokendestination "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/NativeTokenDestination"
+	exampleerc20 "github.com/ava-labs/teleporter/abi-bindings/go/Mocks/ExampleERC20"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	deploymentUtils "github.com/ava-labs/teleporter/utils/deployment-utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func ERC20ToNativeTokenBridge(network interfaces.Network) {
+	const (
+		// This test needs a unique deployer key, whose nonce 0 is used to deploy the bridge contract
+		// on each chain. The address of the resulting contract has been added to the genesis file as
+		// an admin for the Native Minter precompile.
+		deployerKeyStr                     = "ca7269c1fe2a5b86884a119aa516b8d5b641670b83aac0ebf9f2d71affcc12e4"
+		ExampleERC20ByteCodeFile           = "./contracts/out/ExampleERC20.sol/ExampleERC20.json"
+		ERC20TokenSourceByteCodeFile       = "./contracts/out/ERC20TokenSource.sol/ERC20TokenSource.json"
+		NativeTokenDestinationByteCodeFile = "./contracts/out/NativeTokenDestination.sol/NativeTokenDestination.json"
+	)
+	var (
+		initialReserveImbalance      = big.NewInt(0).Mul(big.NewInt(1e15), big.NewInt(1e9))
+		valueToSend                  = big.NewInt(0).Div(initialReserveImbalance, big.NewInt(4))
+		intermediateReserveImbalance = utils.BigIntSub(initialReserveImbalance, valueToSend)
+		valueToReturn                = big.NewInt(0).Div(valueToSend, big.NewInt(4))
+
+		ctx                  = context.Background()
+		deployerAddress      = common.HexToAddress("0x539447ab8Be7e927bE8E005663C81ff2AE951337")
+		tokenReceiverAddress = common.HexToAddress("0x4444444444444444444444444444444444444444")
+		burnedTxFeeAddress   = common.HexToAddress("0x0100000000000000000000000000000000000000")
+
+		emptyDestFeeInfo = nativetokendestination.TeleporterFeeInfo{
+			FeeTokenAddress: common.Address{},
+			Amount:          common.Big0,
+		}
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetA := subnets[0]
+	subnetB := subnets[1]
```

Comment:
This can be replaced with `subnetAInfo, subnetBInfo, _ := utils.GetThreeSubnets(network)`